crates/derive/src/stages/attributes_queue.rs

```diff
@@ -152,10 +195,15 @@ where
 #[async_trait]
 impl<P, AB> ResettableStage for AttributesQueue<P, AB>
 where
-    P: AttributesProvider + OriginProvider + Send + Debug,
+    P: AttributesProvider + PreviousStage + Send + Debug,
     AB: AttributesBuilder + Send + Debug,
 {
-    async fn reset(&mut self, _: BlockInfo, _: &SystemConfig) -> StageResult<()> {
+    async fn reset(
+        &mut self,
+        block_info: BlockInfo,
+        system_config: &SystemConfig,
+    ) -> StageResult<()> {
+        self.prev.reset(block_info, system_config).await?;
```

Comment:
qq - is it ever significant the order in which stages are reset? We're resetting the prev stage recursively before updating the current stage in a few places.

Reply:
So that's always the order in which the reference implementation performs the reset.

Essentially, the way the reference implementation performs a reset is in the following order:
a). the engine queue.
b). resets iteratively up the pipeline and resets each stage from l1 traversal to the attributes queue.

See the order of the `stages` list in the reference implementation here: https://github.com/ethereum-optimism/optimism/blob/6f99544c57ef837afaa96f979a6154a4e8a3604a/op-node/rollup/derive/pipeline.go#L93-L96